src/Symfony/Component/Validator/Constraints/ChoiceValidator.php

```diff
@@ -32,10 +32,14 @@ class ChoiceValidator extends ConstraintValidator
      */
     public function validate($value, Constraint $constraint)
     {
-        if (!$constraint->choices && !$constraint->callback) {
+        if (in_array($constraint->choices, array(null, false), true) && !$constraint->callback) {
```

Comment:
Does it make sense for the user to allow `false` in order to "disable" the `choices` option in favor of `callback` one ? Or should we just check `null === $constraint->choices` ?